core/shared/src/main/scala/fs2/interop/flow/StreamSubscriber.scala

```diff
@@ -66,9 +66,9 @@ private[flow] final class StreamSubscriber[F[_], A] private (
     * since they are always done on the effect run after the state update took place.
     * Meaning this should be correct if the Producer is well-behaved.
     */
-  private var inOnNextLoop: Boolean = _
+  private var inOnNextLoop: Boolean = false
   private var buffer: Array[Any] = null
-  private var index: Int = _
+  private var index: Int = 0
```

Comment:
AFAIK, these are equivalent, but just in case, I made sure `inOnNextLoop` starts as `false`, the `index` shouldn't matter but just for consistency; I also thought of using `-1` but that feels confusing.